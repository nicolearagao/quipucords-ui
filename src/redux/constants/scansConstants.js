const ADD_SCAN = 'ADD_SCAN';
const ADD_START_SCAN = 'ADD_START_SCAN';
const UPDATE_SCANS = 'UPDATE_SCANS';
const DELETE_SCAN = 'DELETE_SCAN';

const GET_SCANS = 'GET_SCANS';
const GET_SCAN_CONNECTION_RESULTS = 'GET_SCAN_CONNECTION_RESULTS';
const GET_SCAN_INSPECTION_RESULTS = 'GET_SCAN_INSPECTION_RESULTS';
const GET_SCAN_JOB = 'GET_SCAN_JOB';
const GET_SCAN_JOBS = 'GET_SCAN_JOBS';

const START_SCAN = 'START_SCAN';
const CANCEL_SCAN = 'CANCEL_SCAN';
const PAUSE_SCAN = 'PAUSE_SCAN';
const RESTART_SCAN = 'RESTART_SCAN';

const EDIT_SCAN_HIDE = 'EDIT_SCAN_HIDE';
const EDIT_SCAN_SHOW = 'EDIT_SCAN_SHOW';
const MERGE_SCAN_DIALOG_HIDE = 'MERGE_SCAN_DIALOG_HIDE';
const MERGE_SCAN_DIALOG_SHOW = 'MERGE_SCAN_DIALOG_SHOW';

const SELECT_SCAN = 'SELECT_SCAN';
const DESELECT_SCAN = 'DESELECT_SCAN';
const EXPANDED_SCAN = 'EXPANDED_SCAN';
const NOT_EXPANDED_SCAN = 'NOT_EXPANDED_SCAN';

const RESET_ACTIONS = 'RESET_SCAN_ACTIONS';

const scansTypes = {
  ADD_SCAN,
  ADD_START_SCAN,
  UPDATE_SCANS,
  DELETE_SCAN,
  GET_SCANS,
  GET_SCAN_CONNECTION_RESULTS,
  GET_SCAN_INSPECTION_RESULTS,
  GET_SCAN_JOB,
  GET_SCAN_JOBS,
  START_SCAN,
  CANCEL_SCAN,
  PAUSE_SCAN,
  RESTART_SCAN,
  EDIT_SCAN_HIDE,
  EDIT_SCAN_SHOW,
  MERGE_SCAN_DIALOG_HIDE,
  MERGE_SCAN_DIALOG_SHOW,
  SELECT_SCAN,
  DESELECT_SCAN,
  EXPANDED_SCAN,
  NOT_EXPANDED_SCAN,
  RESET_ACTIONS
};

export {
  scansTypes as default,
  scansTypes,
  ADD_SCAN,
  ADD_START_SCAN,
  UPDATE_SCANS,
  DELETE_SCAN,
  GET_SCANS,
  GET_SCAN_CONNECTION_RESULTS,
  GET_SCAN_INSPECTION_RESULTS,
  GET_SCAN_JOB,
  GET_SCAN_JOBS,
  START_SCAN,
  CANCEL_SCAN,
  PAUSE_SCAN,
  RESTART_SCAN,
  EDIT_SCAN_HIDE,
  EDIT_SCAN_SHOW,
  MERGE_SCAN_DIALOG_HIDE,
  MERGE_SCAN_DIALOG_SHOW,
  SELECT_SCAN,
  DESELECT_SCAN,
  EXPANDED_SCAN,
  NOT_EXPANDED_SCAN,
  RESET_ACTIONS
};
