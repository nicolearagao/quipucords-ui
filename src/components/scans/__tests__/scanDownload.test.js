import React from 'react';
import { ScanDownload } from '../scanDownload';
import { apiTypes } from '../../../constants/apiConstants';

describe('ScanDownload Component', () => {
  it('should render a basic component', async () => {
    const props = {
      job: {
        [apiTypes.API_RESPONSE_JOB_ID]: 1,
        [apiTypes.API_RESPONSE_JOB_NAME]: 'lorem ipsum'
      }
    };

    const component = await shallowComponent(<ScanDownload {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should handle custom children', async () => {
    const props = {
      job: {
        [apiTypes.API_RESPONSE_JOB_ID]: 1,
        [apiTypes.API_RESPONSE_JOB_NAME]: 'lorem ipsum'
      }
    };

    const component = await shallowComponent(<ScanDownload {...props}>lorem ipsum</ScanDownload>);
    expect(component).toMatchSnapshot('custom');
  });

  it('should have an optional tooltip', async () => {
    const props = {
      job: {
        [apiTypes.API_RESPONSE_JOB_ID]: 1,
        [apiTypes.API_RESPONSE_JOB_NAME]: 'lorem ipsum'
      },
      tooltip: 'Lorem ipsum dolor sit'
    };

    const component = await shallowComponent(<ScanDownload {...props} />);
    expect(component).toMatchSnapshot('tooltip');
  });
});
