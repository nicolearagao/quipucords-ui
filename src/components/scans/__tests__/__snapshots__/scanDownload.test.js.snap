// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanDownload Component should handle custom children: custom 1`] = `
<Button
  onClick={[Function]}
  title="t(table.label_action_scan, {"context":"download"})"
>
  lorem ipsum
</Button>
`;

exports[`ScanDownload Component should have an optional tooltip: tooltip 1`] = `
<Tooltip
  content="Lorem ipsum dolor sit"
  delayShow={500}
  id={null}
  isPopover={false}
  placement="top"
>
  <Button
    onClick={[Function]}
    title="t(table.label_action_scan, {"context":"download"})"
  >
    t(table.label_action_scan, {"context":"download"})
  </Button>
</Tooltip>
`;

exports[`ScanDownload Component should render a basic component: basic 1`] = `
<Button
  onClick={[Function]}
  title="t(table.label_action_scan, {"context":"download"})"
>
  t(table.label_action_scan, {"context":"download"})
</Button>
`;
